Utilities/rFile.cpp

```diff
@@ -2,325 +2,622 @@
 #include "Log.h"
 #pragma warning(disable : 4996)
 #include <wx/dir.h>
-#include <wx/file.h>
 #include <wx/filename.h>
 #include "rFile.h"
-#include "errno.h"
 
 #ifdef _WIN32
 #include <Windows.h>
 
-// Maybe in StrFmt?
-std::wstring ConvertUTF8ToWString(const std::string &source) {
-	int len = (int)source.size();
-	int size = (int)MultiByteToWideChar(CP_UTF8, 0, source.c_str(), len, NULL, 0);
-	std::wstring str;
-	str.resize(size);
-	if (size > 0) {
-		MultiByteToWideChar(CP_UTF8, 0, source.c_str(), len, &str[0], size);
+#define GET_API_ERROR static_cast<u64>(GetLastError())
+
+std::unique_ptr<wchar_t> ConvertUTF8ToWChar(const std::string& source)
+{
+	const size_t length = source.size() + 1; // size + null terminator
+
+	const int size = source.size() < INT_MAX ? static_cast<int>(length) : throw std::length_error(__FUNCTION__);
+
+	std::unique_ptr<wchar_t> buffer(new wchar_t[length]); // allocate buffer assuming that length is the max possible size
```

Comment:
shouldn't this and the return type be `std::unique_ptr<wchar_t[]>` and not `std::unique_ptr<wchar_t>` since you would want the deleter to be `delete[]` and not `delete`